xchange-stream-core/src/main/java/info/bitrich/xchangestream/core/StreamingExchange.java

```diff
@@ -26,6 +28,16 @@
      */
     boolean isAlive();
 
+    /**
+     * Observable for reconnection failure event.
+     * When this happens, it usually indicates that the server or the network is down.
+     *
+     * @return Observable with the exception during reconnection.
+     */
+    default Observable<Throwable> reconnectFailure() {
+        throw new NotYetImplementedForExchangeException();
```

Comment:
Maybe worth just adding a comment here saying that any `NettyStreamingService` backed exchanges can easily implement this method just by calling `subscribeReconnectFailure()`.

Or implement them all of course ;)